test-data/unit/check-statements.test

```diff
@@ -410,6 +410,12 @@ raise object # E: Exception must be derived from BaseException
 raise f # E: Exception must be derived from BaseException
 [builtins fixtures/exception.pyi]
 
+[case testRaiseExceptionType]
+import typing
+x = None # type: typing.Type[BaseException]
+raise x
```

Comment:
Could you also add a negative test (e.g., `x: Type[int]; raise x` still errors)?

Reply:
done